gym_xiangqi/envs/xiangqi_env.py

```diff
@@ -171,15 +259,22 @@ def get_possible_actions_by_piece(self, piece_id):
         return:
             actions that are can be taken by the piece.
         """
-        self.get_possible_actions()
+        if piece_id > 0:
+            possible_actions = self.agent_actions
```

Comment:
Question: shouldn't we call `get_possible_actions()` first which updates the action list, then only assign `possible_actions` to the corresponding action list?

Reply:
Either way works because Python assigns `possible_actions` a reference to the numpy array. Therefore, under the hood, it will be pointing to the updated actions list. Still I think this is a good point, I will change the order to increase readability. 👍👍